Updated the switch portlib to v2.28.1

giroletm opened the pull request:
Ported all of the patches from the switch portlib to the latest SDL2 release (because the current version is from 2020).

I have included ideas to avoid loosing commit history in a pull request I made on the pacman-package repo (because I sent my edits there first): https://github.com/devkitPro/pacman-packages/pull/305

I can provide a build if needed, compiled on my computer with the latest devkitPro at the time of writing. Tested it both on Ryujinx and on my own Nintendo Switch.

Included all contributors in the commit message to avoid their names being lost due to the update if merged.

i'd advise pulling ``main`` from upstream as well (**with tags**) to have the most recent release tags and have an updated ``main`` to be consistent with this update

rsn8887 commented:
This seem to break uae4all2 sound semaphores. Has this been tested with audio and sdl_mixer, etc?

rsn8887 commented:
@cpasjuste can you take a look? Any suggestion how I could fix it? Some include or compiler directive etc?

Sound works but is very choppy and fps goes to 60 instead of 50. It behaves as if the semaphore is completely ignored.

Cpasjuste commented:
I'm on it.

rsn8887 commented:
Yup, fixed! Thank you!!!

Commit:
Stop beep when running iOS apps on ARM-based Macs

(cherry picked from commit bbf38bbbc334bc0ee90d3eec228a968d8fc1689e)

ericwa committed:
Handle DPI scaling in SDL_GetWindowSurface

Fixes DPI awareness of testdrawchessboard (previously, the surface was
being created in points instead of pixels, resulting in the demo app
only drawing in a corner of the screen on High-DPI displays)

*_CreateWindowFramebuffer()/*_UpdateWindowFramebuffer(): are updated
to use SDL_GetWindowSizeInPixels instead of SDL_GetWindowSize() or
window->w/window->h.

Most of the _CreateWindowFramebuffer backends are untested except
for Windows.

Fixes #7047

(cherry picked from commit 67c91353e01f6f2c0cc80c17eeddbad6def7cb01)

slouken committed:
Added support for the Turtle Beach REACT-R and Recon Xbox controllers

(cherry picked from commit 5fded632d6fa532ac25e76ad9a676c9ea726dd02)

slouken committed:
Fixed the share button on the ThrustMaster eSwap PRO Controller Xbox

(cherry picked from commit e4face7c1d41698067b9afdd8c7f0a8f76774da2)

sezero committed:
SDL_x11dyn.h: include XInput2.h also for Xfixes

Fixes https://github.com/libsdl-org/SDL/issues/7343

elmarco committed:
windows/gles: correct indentation

Signed-off-by: Marc-André Lureau <marcandre.lureau@redhat.com>
(cherry picked from commit 654965a6286ccb3ef861310d0cff578788641091)

elmarco committed:
windows/gles: do not unload EGL when context is destroyed

It's legitimate to have multiple contexts.

Signed-off-by: Marc-André Lureau <marcandre.lureau@redhat.com>
(cherry picked from commit 93861e1de00a76080fa6ed756a6bd9c764b3ec07)

openglfreak committed:
x11: Fix duplicate Xinput2 event reception

Passing True for owner_events in the XGrabPointer call makes all
XI_RawMotion events appear in the queue twice, with the only difference
between them being the value of XGenericEventCookie::cookie. These have
always been filtered out by a check in the XI_RawMotion handler,
however with a mouse that polls at more than 1 kHz frequency, there
also exist legitimate events that appear indistinguishable from these
duplicated events. These must not be filtered out, otherwise the
pointer may move at an inconsistent speed, appearing like a bad pointer
acceleration implementation.

Change owner_events to False in the XGrabPointer and remove the
duplicate event detection code to fix this.

Signed-off-by: Torge Matthies <openglfreak@googlemail.com>
(cherry picked from commit f18b5656f6f859e4d4e096d290afd9fae884a5b8)

slouken committed:
Added other names for the DragonRise Inc. Wired Wheel (thanks @felimwhiteley!)

(cherry picked from commit eede2066fbcc342e96592fbbbfe94b2420feb524)

madebr committed:
Fix -Wunused-function warning when configuring with -DSDL_X11_XINPUT2=OFF

lephilousophe committed:
Fix window size in X11 when window manager refuses to resize

(cherry picked from commit a67ae8eed57f4545c739e6eb3a4b8c1169d3afbc)

clbr committed:
Fix GetGlobalMouseState when xi2 is not available at runtime

(cherry picked from commit 0123d6311d5c0059d617324426ef7eef8d577d9b)

slouken committed:
Fixed typo (thanks @Iniquitatis!)

(cherry picked from commit ced02319a6bd001c1c2ce85a48887efe3a7b79d8)

icculus committed:
include: Fix symbols in public headers that start with an underscore.

Fixes #7352.

slouken committed:
cocoa/gles: do not unload EGL when context is destroyed

(cherry picked from commit e8091b89830c3b9bb19eff76463a6e2525f05fa5)

slouken committed:
Added support for the Razer Wolverine V2 controllers

(cherry picked from commit 4994654d4f2d5c96a0faa972ebbb6c653f939d56)

slouken committed:
Fixed rumble caps for the Nintendo Joy-Con controllers

(cherry picked from commit 02cea5fca107991774792c431efc96319ef2d060)

slouken committed:
Removed duplicated conditional

(cherry picked from commit 1a15d506a69d5d73a53728c569ef84a69db51833)

icculus committed:
resampler: Work at double precision on x86-64 machines.

We get audio artifacts if we don't work at the higher precision, but
this is painful on CPUs that have to use a software fallback for this,
so for now (that is, until we have a better solution), get better output
on amd64 chips, where the cost is less painful.

(cherry picked from commit 1e5e8e2fda3796e76e6f7b1c39683925a3e9fed9)

slouken committed:
Fixed the packet size for Xbox One controllers

(cherry picked from commit 021a7cfa13c20d518ebfb6e96c3305150256983b)

slouken committed:
Rumbling the Google Stadia Controller over Bluetooth works on Linux and macOS

(cherry picked from commit 4aeec9d8c2ad2fa0465999ecb4fd5c20734f1b27)

slouken committed:
De-duplicate Google Stadia controllers on macOS

(cherry picked from commit 87a83787a3a0a9922b02b35ba809d9da86930fc8)

sezero committed:
fix MSVC build failures in CI after commit 1e5e8e2f

(cherry picked from commit a2f8cf8fc0c6f4fe8a904284060f61f19f129efa)

slouken committed:
Fixed view frame from flipping back and forth between landscape and portrait on iOS

(cherry picked from commit b2d913883c39e0a53e55eb9286d7e442d565554e)

slouken committed:
Added support for the Backbone One PlayStation Edition

(cherry picked from commit 59bf1c2853d5b06295c939099f6cf5f1611f70b5)

slouken committed:
Changed the VID/PID used for the Backbone One controller on iOS

(cherry picked from commit 516e48515e32f74aa98a3df4cbdb23b9c477f2f0)

icculus committed:
wikiheaders: Work to make the wiki exist primarily in Markdown format.

This adds a means to mass-convert the whole wiki to Markdown as a one-time
operation, and then some fixes to make --copy-to-headers correctly deal with
Markdown-formatted wiki pages.

(cherry picked from commit 936a51d5cc50afd76e291acd36bcb95513ce8919)

icculus committed:
wikibridge: Fixes for manpage generation from Markdown format.

(cherry picked from commit 5ff49955ab71121ffbc49eb6a7be69707cbd90f3)

icculus committed:
wikiheaders: fixed see-also conversion

(cherry picked from commit d748a454a86d6250b724fe94066eb98fbf4eea05)

icculus committed:
Sync wiki -> headers.

slouken committed:
Fixed relative mouse motion over remote desktop

Setting the cursor clip area to a single pixel prevents the relative mouse motion remote desktop warping from working, so the mouse is never recentered.

(cherry picked from commit daffe02b117ccd484763eadc716e4d4453c89868)

icculus committed:
wikiheaders: Fix wikilinks inside code sections a little. Not perfect yet.

(cherry picked from commit 5b0351a672701e28c3bfb17d44a458b82e8a4175)

devnexen committed:
USE_DEV_RANDOM close the file descriptor even in the rare case it can't read it

slouken committed:
Added mapping for DualSense Edge Wireless Controller on Linux

This is a mapping when using evdev, not the more advanced HIDAPI support.

(cherry picked from commit 206fa4dafb633ed074b930cbb3440a9925c532f2)

icculus committed:
wikiheaders: fixed wikilinks in Markdown code sections.

(cherry picked from commit a4796334550da0963ed957829f41b9499a472d73)

slouken committed:
Fixed locking up the Logitech F310 with the PlayStation controller detection

(cherry picked from commit da134a30396e12786c14fe8d1190ab05c67d9dba)

icculus committed:
wikiheaders: Bridge README files in the docs directory to wiki.

Fixes #6026.

(cherry picked from commit 2506676f34dc9d3e64ab7bbd9c947b8c13364913)

eloj committed:
Always allocate zt in output of SDL_iconv_string()

Before this, the function could not be used on buffers,
as it would not account for the zero-termination unless
it was included in the input.

(cherry picked from commit 5f5abb680523e1adedd3fca2a8e252db01fc1c52)

icculus committed:
wikiheaders: ignore wiki's README/FrontPage.md

(cherry picked from commit 6d1e14b7924e45937941a64bcaf751e17b41a8e9)

slouken committed:
Fixed SDL windows getting the maximized state after leaving fullscreen on macOS

(cherry picked from commit 2df08fb1b27b721998a1380a2acdbe1494b5591e)

icculus committed:
wikiheaders: Use Windows endlines in the source, Unix in the wiki.

(cherry picked from commit 7745c9b3ae3f2f47e2b3e55399afc5e7b86c0102)

icculus committed:
wikiheaders: Sort pages before listing them in README/FrontPage.md.

(cherry picked from commit 99c38268cd1325bb64739217adbeebaae7ae6a84)

icculus committed:
Sync wiki -> headers.

icculus committed:
docs: Removed bogus file from wiki bridge.

That was fixed to not do that again by https://github.com/libsdl-org/ghwikipp/commit/7e429a13d385e448dd2efaf014e5aaddb6eff66b

slouken committed:
Follow the same pattern for releasing the window data in the UIKit driver

Possible fix for https://github.com/libsdl-org/SDL/issues/7361

(cherry picked from commit ebdb32065163b8ee1ef9c1a0dfa601335b4ad83d)

madebr committed:
cmake: use compatible interface properties to disallow linking to a different version of SDL

slouken committed:
Don't clobber an existing entry with a different version when adding a gamepad mapping

(cherry picked from commit 727c7d4e2f44102bb571180929bfbb5d527f4f27)

slouken committed:
Fixed typo in previous cherry-pick

icculus committed:
dataqueue: Removed non-public SDL_ReserveSpaceInDataQueue function

This wasn't used, and it was just asking for trouble.

(cherry picked from commit f833e005e10f63979fd545d8bc29b4f1a7c36486)

icculus committed:
dataqueue: Make thread safe.

Each data queue gets its own mutex and each function obtains it.

Fixes #7390.

(cherry picked from commit 8b9a9384132e39a49a14cf41b292018ce4f4f251)

sezero committed:
dynapi: replace uses of deprecated SDL_assert_state and SDL_assert_data

hramrach committed:
evdev_kbd: Use current keymap

keymap can change over time, caching the keymap causes wrong keys
returned when user changes keymap during runtime

Signed-off-by: Michal Suchanek <msuchanek@suse.de>

divVerent committed:
Fix Xbox One gamepad axis assignment on SDL_JOYSTICK_ANDROID API (#7405)

(cherry picked from commit 6f1f586086a18ab32ddccc05e4616aac02774a1b)

slouken committed:
Added mapping for the Xbox Elite Core controller connected via Bluetooth on Android

(cherry picked from commit d08338ddc5bb3cae1e4ce06201a2cd3c53215aa0)

slouken committed:
Fixed which window we check flags for when updating fullscreen mode

slouken committed:
If we're not the current fullscreen window and we don't want to be fullscreen, don't touch the current video mode

Fixes an assertion in qemu on macOS, which creates multiple hidden windows:
2023-03-07 09:56:55.304 qemu-system-i386[9931:2984587] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'NSWindowStyleMaskFullScreen cleared on a window outside of a full screen transition. Called from (
	0   AppKit                              0x00000001968f77ac __25-[NSWindow setStyleMask:]_block_invoke + 140
	1   AppKit                              0x00000001968f76cc NSPerformVisuallyAtomicChange + 108
	2   AppKit                              0x00000001968f7580 -[NSWindow setStyleMask:] + 188
	3   libSDL2-2.0.0.dylib                 0x0000000106531328 Cocoa_SetWindowFullscreen + 568
	4   libSDL2-2.0.0.dylib                 0x0000000106501700 SDL_UpdateFullscreenMode + 1316
	5   libSDL2-2.0.0.dylib                 0x0000000106504270 SDL_SetWindowFullscreen_REAL + 328
	6   libSDL2-2.0.0.dylib                 0x000000010650117c SDL_FinishWindowCreation + 100
	7   libSDL2-2.0.0.dylib                 0x0000000106500998 SDL_CreateWindow_REAL + 1812
	8   libSDL2-2.0.0.dylib                 0x00000001063efd54 SDL_CreateWindow + 76
	9   qemu-system-i386                    0x00000001050aa600 sdl2_window_create + 192
	10  qemu-system-i386                    0x00000001050a9fa0 sdl2_2d_switch + 196
	11  qemu-system-i386                    0x0000000104e9c784 displaychangelistener_display_console + 524
	12  qemu-system-i386                    0x0000000104e9e36c register_displaychangelistener + 264
	13  qemu-system-i386                    0x00000001050ab6d0 sdl2_display_init + 304
	14  qemu-system-i386                    0x000000010505c870 qemu_init + 13380

slouken committed:
Use SDL_GL_DeleteContext instead of Cocoa_GL_DeleteContext for context cleanup

That will make the current context NULL before deleting the context.

slouken committed:
Disassociate the SDLOpenGLContext from the view before deleting it

If we don't do this, the view will be blanked even if another context is current and rendering from that context won't be visible.

Fixes https://github.com/libsdl-org/SDL/issues/4986

sezero committed:
tests: added -DHAVE_SIGNAL_H to CFLAGS in watcom makefiles

slouken committed:
Fixed uninitialized variable

sezero committed:
fix SDL_FindFreePlayerIndex so it can really return -1.

deveee committed:
Fixed incorrect modifier keys handling on macOS

(cherry-picked from commit 45a58b7882a253db29b1b1393bc2a1fe030d5955)

Kontrabant committed:
wayland: Always use a scale factor of 1 for exclusive fullscreen emulation

Lokathor committed:
Make mappings for "Retrolink SNES Controller" with `SDL_GAMECONTROLLER_USE_BUTTON_LABELS` both on and off

I've only tested this on windows, but I went ahead and made the same changes for linux and mac because I assumed it's the same there and that we'd want to keep the three platforms in sync.

(cherry picked from commit b8bc4a234b93efa7a28c7fd7d524d905d9b76e6d)

Kontrabant committed:
tests: Always set valid dimensions for the fullscreen mode

slouken committed:
Fixed the documentation for SDL_SetRelativeMouseMode()

(cherry picked from commit fb0c3197e0e4696551822815bdce0e187a0b06fc)

Qrox committed:
Uses integer arithmetics in SDL_ResampleAudio

- Avoids precision loss caused by large floating point numbers.
- Adds unit test to test the signal-to-noise ratio and maximum error of resampler.
- Code cleanup

(cherry-picked from commit 20e17559e545c5d3cfe86c1c4772365e70090779)

sezero committed:
Removed guard around the enabled assertions support code

Otherwise, SDL_assert_always() wouldn't work.

Fixes https://github.com/libsdl-org/SDL/issues/7433

slouken committed:
If the client rect is empty, use the last known window size

This happens on Windows 11 with fullscreen desktop windows when the desktop is brought up with the Windows+D shortcut.

Fixes https://github.com/libsdl-org/SDL/issues/7419

(cherry picked from commit 2ca727aec6f5f264620f80999beb5ef77eefec4a)

slouken committed:
Code style: changed "sizeof foo" to "sizeof(foo)" (thanks @sezero!)

(cherry picked from commit c6443d86c92e962683a1efe5f123a144988875b5)

slouken committed:
Wait for the GCController framework to see IOKit devices

It occasionally takes a few millseconds for the GCController framework to handle the device notification and set up the device

Fixes the duplicate controller issue in https://github.com/libsdl-org/SDL/issues/6686

(cherry picked from commit 645823fc901c98b688512d3b3e70cc8922e8140c)

slouken committed:
Removed a duplicate delay accidentally added during code refactor

(cherry picked from commit 926db1bd544364a9e4ab61e60bcfbe75e05adf56)

slouken committed:
Added header for SDL_Delay()

sezero committed:
fix bad merge of sizeof style patch

skitt committed:
Specify bash for androidbuildlibs.sh

This script relies on string indexes in parameter expansions, which
aren't suppored by /bin/sh (e.g. dash).

Based on a patch by Roflcopter4:
https://github.com/joncampbell123/dosbox-x/pull/3850

Signed-off-by: Stephen Kitt <steve@sk2.org>
(cherry picked from commit bbfd5b3fb2eed8f95919febf3bc0fec4bc605cbe)

slouken committed:
Fixed build warning on Android

(cherry picked from commit 2af6f4a3e4f0a82ccbb7ce142c914b5aa5df2000)

slouken committed:
Removed space between sizeof and parentheses

slouken committed:
Added support for the trigger buttons on the Victrix Pro FS for PS5

(cherry picked from commit 8d7e61ea42364c256b76e7f6f2b6d6f90742bcb7)

divVerent committed:
Android: indicate gamepaddb entries where axis order changed.

Axis order was changed in 6f1f586086a18ab32ddccc05e4616aac02774a1b to improve
default mappings.

(cherry picked from commit de3909a190f6e1a3f11776ce42927f99b0381675)

slouken committed:
Ignore the PS4 packet CRC if it's not being set correctly

This fixes handling the 8BitDo SN30 Pro with the 2.00 firmware in PS4 mode

Fixes https://github.com/libsdl-org/SDL/issues/7270

(cherry picked from commit 3951cae4a56fddcb42ce4141a6518c97fa9125cd)

madebr committed:
cmake: get_filename_component(EXT) retturns extension including leading dot

1bsyl committed:
Refresh generation of SDL_blit_auto.c

1bsyl committed:
Simplify SDL_BLENDMODE_MUL

1bsyl committed:
Simplify SDL_BLENDMODE_MUL in sdlgenblit.pl

1bsyl committed:
Refresh generation of SDL_blit_auto.c after simplifying sdlgenblit.pl

1bsyl committed:
Change SDL_BLENDMODE_MUL for gl renderers
 Add FIXME for PSP and DirectFB

slouken committed:
Added gamepad mapping for Flydigi Vader 2 with the latest firmware (6.0.4.9)

(cherry picked from commit ed0937598e0a637a1033c90a1b31886537bb288c)

icculus committed:
wasapi: Force WIN_CoInitialize duing device open.

Otherwise this won't work if you open a device on a background thread.

Fixes #7478.

ivan-mogilko committed:
Fixed SDL_RenderSetVSync does not update flags if simulated vsync is on

This lets the user to correctly detect current vsync state by reading SDL_RendererInfo.
Also fixes SetVSync's return value check (it may be positive for error too).

madebr committed:
testsurround: fix channel names

Used table in include/SDL3/SDL_audio.h as reference.

(cherry picked from commit e1b8a032391f4b24453d5bcbe725a35b3894ea5d)

madebr committed:
video/blit: fix -Wimplicit-fallthrough warnings

(cherry picked from commit f7961b7c9ad598b031cd033f19e93c6d882e8e41)

madebr committed:
testautomation_audio: fix -Wimplicit-fallthrough warning

(cherry picked from commit 11c70406c322b5686f35349acc9659c4d8d5a8f7)

madebr committed:
video: fix warnings about unused variables when building without duff's loop

(cherry picked from commit 0c3777d5661dd0bf60ba3db634f79c2121371092)

skitt committed:
Split up CFLAGS construction so it can be overridden

CFLAGS currently takes care of two aspects of the build: optimisation
and debug symbols on one hand, the include path on the other. The
former should be overridable by users, the latter shouldn't.

This patch moves the include path flag to CPPFLAGS, which is
appropriate for pre-processor directives. This leaves CFLAGS with only
overridable flags.

Based on a patch by Roflcopter4:
https://github.com/joncampbell123/dosbox-x/pull/3850

Signed-off-by: Stephen Kitt <steve@sk2.org>

madebr committed:
testautomation_math: avoid equality tests with INFINITY

Fixes this warning:
 warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]

madebr committed:
cpuinfo: use __cpuidex instead of __cpuid

The classic Intel Compiler does not clear the ecx register prior
to executing the cpuid opcode.

madebr committed:
byteswap: Don't use intrinsic byteswap functions with Intel C compiler

It reroutes intrinsic _byteswap_u(short|long|int64) call to libc function.

madebr committed:
cmake: new LLVM based Intel compiler does not recognize MSVC's /MP

madebr committed:
cmake: increase threshold for stack probes

The /Gs argument controls the number of bytes that local variables
can occupy before a stack probe is initiated.
By setting it to a huge value, no calls to __chkstk are inserted.

This change is needed for the classic Intel C compiler to build SDL
with -DSDL_LIBC=OFF.

madebr committed:
cmake: add support for building with Intel C compiler

madebr committed:
ci: test with (old) Intel compiler + (new) oneAPI compiler

madebr committed:
cmake: add /Q_no-use-libirc flag when building a no-libc library

The new Intel LLVM library needs this because when building SDL2
in release mode.

madebr committed:
Implement _intel_fast_(memcpy|memset)

The classic Intel compiler generates calls to these functions when
building the SDL library with SDL_LIBC=OFF.

madebr committed:
cmake: disable warnings in libm + warning about EMMS instruction

SDL_BlitCopyMMX ends with _mm_empty(), so the MMX state should be emptied.

(_mm_empty is the intrinsic function for emms)

FtZPetruska committed:
Fix clang-tidy error on LLVM 16.

Starting LLVM 16, clang-diagnostic-implicit-function-declaration is
treated as an error.
(cherry picked from commit ea093378a29b786edbaa265045638f66d7f83d8b)

sezero committed:
wikiheaders.pl: make it work with older perl versions, e.g. 5.10.

1bsyl committed:
backport x11/sdl2 fixes

4b1378f

X11: fix size/position (test video_setWindowCenteredOnDisplay)

this fix x11 backend to correctly pass video_setWindowCenteredOnDisplay()

    get border values early (eg status bar)
    wait for size/position change to get valid values

d4d26e0

testautomation_video: if SDL_SetWindowSize/Position isn't honored, we should check there is an event
x11: send the events if various occasions

bdach committed:
Android: Fix missing type in SDLControllerManager

The missing type in question is causing compilation failures.
The error was introduced in dca3fd8307c2c9ebda8d8ea623bbbf19649f5e22,
which was a backport of commit de3909a190f6e1a3f11776ce42927f99b0381675
from SDL3 to SDL2.

Because `int nballs` was removed as a parameter from the controller
API in SDL3 in revision fcafe40948fe308cc9552df5a3d625ee2725de5a,
this change is only applicable to the SDL2 branch.

bdach committed:
Android: Fix mismatching JNI method signature

Another regression from commit dca3fd8307c2c9ebda8d8ea623bbbf19649f5e22,
which was a backport of commit de3909a190f6e1a3f11776ce42927f99b0381675
from SDL3 to SDL2. This time the regression is much less apparent,
however, due to two reasons:

- It only appears when the SDL project is ran on an actual device
  due to magic stringage.

- More importantly, the regression was partially hidden due to
  the nature of de3909a190f6e1a3f11776ce42927f99b0381675.

  The commit which was backported added a single `I` to the JNI method
  signature in `SDL_android.c`, representing the added `int axis_mask`
  parameter. The parameter was added to both SDL2 and SDL3.

  However, notably, that `I` was added *after* commit
  fcafe40948fe308cc9552df5a3d625ee2725de5a, which removed the
  `int nballs` parameter from the joystick API, but only from the SDL3
  branch.

  Therefore, in totality, what should really have been a merge conflict,
  was obscured by the fact that the SDL3 branch ended up having
  a net-identical JNI signature to the SDL2 branch, due to having
  one bool param removed and one added - while, in fact, the SDL2 branch
  needed one bool param added and none removed.

kallehed committed:
Made the Android documentation more clear and understandable (#7535)

(cherry picked from commit 11142ec2562cc3cce1ea2e09e9758d8b6cfc07bb)

madebr committed:
SDL_blit_copy: Don't call potentially FPU using SDL_memcpy in SDL_memcpyMMX

Kontrabant committed:
events: Increase the size of the name buffer when logging events

Event names have grown in length and are occasionally truncated when being logged (e.g. SDL_EVENT_WINDOW_PIXEL_SIZE_CHA). Increase the event name buffer size to handle the longer names.

(cherry picked from commit 203a2a76fc0af2094de84d75dbf959c7db73fe72)

capehill committed:
testgles2_sdf: Call correct function to get shader info log

(cherry picked from commit b8c88cc58423b7e654b18fdaf1aab5a343114586)

FtZPetruska committed:
SDL_hidapi_steam.c: Fix compilation under c2x.

When N2935 is implemented, the enum breaks compilation. Use a #define of
the SDL booleans instead.

(cherry picked from commit 975039ce0d3587c42e11f2922240957fc8f74166)

slouken committed:
Increased precision for PS4 sensor data conversion

(cherry picked from commit 3340864786314d77e04a0dd2bf1ebffbc43bfc31)

slouken committed:
Fixed the accelerometer sensitivity for the Armor-X Pro controller

(cherry picked from commit 8029a767eef4147c07cde965bd5a2d2e11e237ee)

1bsyl committed:
Fixed bug #7515 - Window still grabbed after destroying

1bsyl committed:
Safety fix: clear "_this->current_glwin" when destroying the window

(cherry picked from commit 0c048d98af449f1bc540e256c90eb792f1effff4)

1bsyl committed:
Safety fix: clear "_this->wakeup_window" when destroying the window

(cherry picked from commit c838ccf0e3a330ebbcf433698b8f5370c6fa80d6)

madebr committed:
Fix remaining -Wimplicit-fallthrough warnings

slouken committed:
Fixed the accelerometer and gyro axes for the Armor-X Pro controller

(cherry picked from commit 37517557ae076cd94acd9ca3b739e1f32c150a43)

slouken committed:
Fixed gyro values for the Armor-X Pro controller

(cherry picked from commit 8df7b4d8539338121b1e34e29f331b28f1f4ff05)

slouken committed:
The Xinmotek Controller is used in multiple products with different mappings

The mapping included here is for the Ultimate Atari Fight Stick

(cherry picked from commit f1099f8e70a1284c2ee3031f2ea0cfc84f22110c)

Kontrabant committed:
x11: Don't allow the changing of certain scancodes

The X11 driver uses scancodes derived from keysyms to map the scancodes for extended keys to the physical keyboard, however, this can be incorrect when using certain XKB options (e.g. caps:swapescape), which changes the keysyms emitted by certain keys, but does not imply that their scancodes or positions should be altered. Mark selected scancodes as being non-remappable so that their scancodes aren't changed by toggling XKB mapping options.

korli committed:
Haiku: use a BLooper for events.

only create a BApplication when it doesn't already exist.

Wohlstand committed:
include: Fixed a C++-style single line comment.

https://github.com/libsdl-org/SDL/commit/78725dc0cd8502a91d2565af44ff9e6e65f0f30d

slouken committed:
Fixed analog triggers on the DualSense controller

(cherry picked from commit a67d41050164e447048fa55179cc31b23be9d9b7)

slouken committed:
Don't probe for PlayStation controllers when we already know the controller type

Fixes https://github.com/libsdl-org/SDL/issues/7556

(cherry picked from commit 655a07bdd88d6bd19fe97c3abd39a2e2eef252ec)

slouken committed:
Fixed crash if dbus can't be initialized

(cherry picked from commit cf2c6fa58dea02edb870cff7090fd69f3dfbd081)

sezero committed:
fix build after commit 22685556e5d9fc2e2a7023075e51b69fe770e64a.

dubet committed:
Fix key code names for ISO keyboard layouts

(cherry picked from commit 60dcaff7eb25a01c9c87a5fed335b29a5625b95b)

slouken committed:
Fixed crash if joystick->hwdata != NULL && device->driver == NULL

This should never happen, but we're seeing it in the wild, so make sure that we can never call into a NULL device driver.

(cherry picked from commit e13b74ccf0b20a20bd4baf1617bdc4f84c13feb7)

slouken committed:
The macOS minimum deployment target is now 10.11

Xcode 14.3 does not allow targeting 10.9, the minimum recommended version is 10.13 and the minimum possible version is 10.11.

(cherry picked from commit 73b2faea4e39aa27b4142f9d4d39f5807a3b3688)

slouken committed:
The latest Armor-X Pro firmware update fixes the sign on the Z axis of the sensors

(cherry picked from commit 558bbbb07ae0afa0108c1560915351cfa13442b7)

slouken committed:
Clarified the expected sensor value when a game controller is at rest

(cherry picked from commit 1a20ccb2897fc9b12ad5242f9128cb6bb60368b1)

slouken committed:
More defensive coding against dangling device pointers

(cherry picked from commit acadb6f873acaa0de094ad10a88f877d540ce0ec)

madebr committed:
cmake: allow users of sdl2-config.cmake to not add -mwindows to the link options

slouken committed:
Report PS3 accelerometer values in m/s2 instead of Gs

(cherry picked from commit 0467301baf57389d96fafbe169fb07f4c10169bc)

szsam committed:
Check return value of SDL_malloc()

(cherry picked from commit a4604cb0d616db97921a2674a63b45f3fdfb81bc)

sezero committed:
hidapi/mac: import mainstream commit c1b9d2ad

https://github.com/libusb/hidapi/commit/c1b9d2ad985c826f7916aa793d4728885a43a0f4
Properly handle the close of run loop on macOS
(https://github.com/libusb/hidapi/pull/522)
- as per documentation `kCFRunLoopRunStopped` should be handled once the runloop is closed via `CFRunLoopStop`;
- if it is not handled - a race condition/crash may happen on the latest macOS when a device gets disconnected while being open;

(cherry picked from commit 62d1a2c836c6176a6c76d54229eafa7d011bb9fb)

sezero committed:
hidapi/mac: replace sprintf uses with snprintf.

Reference issues:
-  https://github.com/libusb/hidapi/pull/509
-  https://github.com/libusb/hidapi/pull/511

(cherry picked from commit ae9119c36bdcf2214d83c6d35c436d22b5fd03cb)

LiquidityC committed:
wayland: Fixes a memory leak wheere primary selection isn't freed

(cherry picked from commit 43f65a6ef08a502485c6107b255a9b22c890acd5)

Kontrabant committed:
wayland: Destroy proxy wrappers and callbacks before event queues

Destroy any proxy wrappers and callbacks before the associated event queues to silence libwayland warnings about destroying the queues while proxies are still attached.

slouken committed:
Added support for the Astro C40 in Xbox 360 mode

(cherry picked from commit a4b4dff4a2998699757f12d29fdd4c632db884fe)

slouken committed:
By default, OpenBSD uses Linux controller mappings

Fixes https://github.com/libsdl-org/SDL/issues/7609

(cherry picked from commit 255c3b7c82968def667da860cbda01f267b4edf1)

slouken committed:
Removed memset_pattern4() from SDL headers

This requires including string.h, which isn't always done, so this commit simplifies dependencies on macOS

Fixes https://github.com/libsdl-org/SDL/issues/3657

LocOlli committed:
Fix bug memory acess

Bitmap is not initialized before use, that cause segmentation fault on function use it. Fix by allocate memory before use.

Commit:
SDL_blit_N.c: Move ppc64le swizzle outside of loop

An in-place swizzle mutation was erroneously inside of a loop, which
caused each consecutive 4-pixel vector to alternate between correct and
incorrect endianness.

The bug was introduced in 715e070d299fc547cce8c52915b75e6316af6bff.

Thanks to RobbieAB for reporting the bug.

Fixes https://github.com/libsdl-org/SDL/issues/3428

(cherry picked from commit 9142292f4ad6d9bc8a2bc7c874e2fde54befeb9b)

slouken committed:
Removed redundant __powerpc__ check

(cherry picked from commit b6df25c33497388f16a4ed2af896ef14277c8865)

icculus committed:
coreaudio: Flush/stop the AudioQueue before disposing of it.

Otherwise, a CoreAudio thread lingers forever, and coreaudiod eats CPU
until the SDL process terminates.

Fixes #7689.

slouken committed:
Changed all variadic macros into fixed (thanks @Wohlstand!)

slouken committed:
Check to make sure the Windows joystick device has buttons and axes

This fixes incorrectly detecting the ROG CHAKRAM X mouse as a game controller on Windows 10

(cherry picked from commit 642504bc5993a6acd4d9c1d4c1314c6b1cf6e4e1)

icculus committed:
blit: Add a case for 8-bit blits that sdl12-compat exposed.

sdl12-compat can get into a state where a color-keyed surface is
marked for blending, but wants to blend with full alpha (which
is the same as _not_ blending), so rather than fail to find a
blitter in that case, it just selects the colorkey blitter.

Reference https://github.com/libsdl-org/sdl12-compat/issues/233

valid-ptr committed:
SDL emscripten port: preventDefault should not be called on mousedown. Otherwise mouseup will not be fired outside iframe in Chrome-based browsers

icculus committed:
dynapi: Accept a comma-separated list of libraries to attempt to load.

First one to load and have the necessary symbol is the one we accept,
if any. Once we accept one, we won't try loading others.

Fixes #7613.

icculus committed:
audio: Load .WAV files with format->blockalign==0.

In theory this is illegal, but legit wavefiles in the field do it, and
it's easy to bump it to 1 for general purposes.

Formats with more specific alignment requirements already check for them
separately.

Fixes #7714.

(cherry picked from commit 2e646c7141b3009628abda2a964ba5f9d1702e1a)

icculus committed:
power: On Linux, compare status strings as case-insensitive.

In case something reports "Device" when we expected "device", etc.

Reference Issue #6835.

icculus committed:
VisualC: turn on multi-processor compilation.

This makes SDL2 build in 12 seconds, instead of 50, on my Windows
workstation, with Visual Studio.

sezero committed:
style fixes for SDL_PROC macros.

madebr committed:
testgles2_sdf: build with CMake + fix build errors/warnings

sezero committed:
test/testgles2_sdf.c: type fixes from commit e26e893daf, style clean-up.

slouken committed:
Moved the controller list out so it can be included elsewhere

(cherry picked from commit 24007b00b2a5e9d09d953db79a24dd51b78e598b)

slouken committed:
Undo name change for the controller list

(cherry picked from commit f60622c5103e37f221aa8252acfa9073cfbafbc7)

TheSpydog committed:
Fix Xbox link error from IsRectEmpty

(cherry picked from commit 376a3cd100a3d59e887496c75a1ac49ab4a2d8ec)

slouken committed:
Make sure the sentinel is at the end of the current event pump cycle

If we're waiting, it's possible to not get any events, then add the sentinel, then pump again and then add another sentinel. We want to make sure we only have one sentinel and that it's at the end of the currently pumped events.

Testing:
* Verified test case in https://github.com/libsdl-org/SDL/issues/6539
* Verified test case in https://github.com/libsdl-org/SDL/issues/5350

Fixes https://github.com/libsdl-org/SDL/issues/6539

slouken committed:
Added the hint SDL_HINT_ENABLE_SCREEN_KEYBOARD to control whether the on-screen keyboard should be shown when text input is active

Fixes https://github.com/libsdl-org/SDL/issues/7160

slouken committed:
Don't map the top keyboard row to numbers when using the one-handed DVORAK layouts (thanks @tormol!)

Fixes https://github.com/libsdl-org/SDL/pull/5127

slouken committed:
Use numeric codes for Android versions, to avoid SDK dependencies

(cherry picked from commit 3f6b2d1a61d0c0dcd6af7cc951c1d8c267ed1526)

slouken committed:
Fixed reporting backspace key if there is no text in the edit buffer (thanks @312937!)

This workaround isn't necessary at API 30 and above.

Fixes https://github.com/libsdl-org/SDL/issues/7039

(cherry picked from commit c971795954bc85086cd20cfc71b45a311b7ba0d1)

slouken committed:
Fixed handling the Saitek P3600, which doesn't use the Xbox 360 controller protocol

(cherry picked from commit cf0abf99118f1ea5ba423ce62950e5840db2c649)

slouken committed:
Fixed duplicate Joy-Con controllers on macOS

If both Apple and HIDAPI drivers see the controller, HIDAPI will be preferred.

Fixes https://github.com/libsdl-org/SDL/issues/7479

(cherry picked from commit 61ef4efdfae7e384b0d24b1b171a9ee6203a3f0d)

Kontrabant committed:
wayland: Validate surfaces and window data before sending touch events

Ensure that incoming touch events originate from valid surfaces owned by SDL and have proper window data before forwarding them to the touch subsystem, or the window focus pointer that is sent with the event may not be a pointer to an SDL window.

Kontrabant committed:
Fix unused variable warnings

slouken committed:
Document the Android SDK versions checked in Java code

(cherry picked from commit b48b1ce50042168c5b9b310fa495cb355903c241)

slouken committed:
Updated source to match SDL function prototype style

slouken committed:
Disable unused local typedefs warning

This triggers on gcc 4.8.4 for compile time asserts inside of functions

Fixes https://github.com/libsdl-org/SDL/issues/7732

Co-authored-by: Ozkan Sezer <sezeroz@gmail.com>

madebr committed:
cmake: allow #include "SDL2/SDL.h.h" when using SDL2 as subproject

icculus committed:
video: Only specify some GL context attributes if not the explicit default.

Just in case it upsets some OpenGL drivers unnecessarily.

Fixes #7730.

(cherry picked from commit d4bc393efebdd45a95430397df106b69b1c26df3)

sezero committed:
fix mingw build failures in CI with clang-tidy.

(cherry-picked from commit a8a72de6fbcf9afe8a6e8760f31a79a0ed6fed1a)

slouken committed:
Added support for printing wide strings using "%ls" syntax

(cherry picked from commit 128ca7016018178c0c3231db7db2005dbf234068)

madebr committed:
wayland: don't use libdecor_frame_get_* functions when header doesn't have it

and we're statically linking to the library.
This fixes building SDL with -DSDL_WAYLAND_SHARED=OFF

(cherry picked from commit 5b5b67df203e9f8391e7e53aa5de54abf14c482e)
(cherry picked from commit e6f635ca172027f57542b715787ac5b50f3f6d33)

icculus committed:
wasapi: Deal with HDMI or DisplayPort-based audio devices.

They can vanish for UP TO EIGHT SECONDS...!

This is for devices that connect to HDMI/DisplayPort/etc, where it
presumably has to wait for a display to get up and running before it
can play audio through it, so one can see the audio device fail when
changing display modes, or the system returning from sleep. Since this
can be triggered by a game changing video resolutions at startup (either
before or after opening the audio device!), it's important to deal with.

In normal conditions, it shouldn't take this long to open or recover an
audio device, but this is better than unexpectedly losing the device
in this situation.

Fixes #7044.
Fixes #5571.

madebr committed:
windows: SDL_immdevice needs SDL_timer.h include

madebr committed:
windows: SDL_Delay expects a 32-bit integer

slouken committed:
Disable SDL_JOYSTICK_RAWINPUT_XINPUT

Apparently when using the Xbox One Wireless Adapter, using XInput at the same time as raw input will cause the controller to turn off immediately after connecting. This appears to be a bug in the Windows 11 driver stack, but since WGI provides all the extended functionality we need, this can be turned off for now.

Fixes https://github.com/libsdl-org/SDL/issues/3468

slouken committed:
Fixed build warning

icculus committed:
windows: Don't allow non-resizable windows to be maximized.

Fixes #6346.

icculus committed:
x11: Attempt to wait for SDL_MaximizeWindow to complete before returning.

Fixes #7070.

sezero committed:
testautomation_guid.c: fix integer warning in 32 bit builds.

a1batross committed:
wayland: reset orientation bitmask before reading values from hint on QtWayland

Signed-off-by: Alibek Omarov <a1ba.omarov@gmail.com>

a1batross committed:
hints: clarify support for comma-separated values for QtWayland orientation that's available since 2.0.22

Signed-off-by: Alibek Omarov <a1ba.omarov@gmail.com>

slouken committed:
Fixed WGI immediately being correlated with raw input devices with no input

The at rest match state is 0x0000008800000000, not 0

(cherry picked from commit e8b5b4881c8eb31bb7de5adef3479305a73fb0bf)

slouken committed:
Don't use raw input while Remote Desktop is active

Raw input will not send game controller events while Remote Desktop is active, so dynamically switch between XInput and raw input when Remote Desktop state changes.

Fixes https://github.com/libsdl-org/SDL/issues/7759

(cherry picked from commit 767507fcf6f4607edb413acb53a6c029863701ed)

slouken committed:
Fixed build if SDL_JOYSTICK_RAWINPUT_MATCHING isn't enabled

(cherry picked from commit 7c55845c804e2272df8bc111dea5d3e9d11670a0)

slouken committed:
Fixed detection of the Steam Virtual Gamepad on macOS

(cherry picked from commit 5fcd70578bdde3eaaa51aa4ab6c24ed08bd49db3)

slouken committed:
Revert "Fixed detection of the Steam Virtual Gamepad on macOS"

This reverts commit 7aa28ce279045884653c8153a5cee99f4622af10.

Steam has been updated to send a version of 1 to avoid conflicts with controllers that report a version of 0.

slouken committed:
Steam uses a different VID/PID for the Steam Virtual Gamepad on Windows

(cherry picked from commit 6815e75cafee8e7caaacf1f428b6539c8bbc7f7d)

slouken committed:
Fixed typo in cherry-pick (thanks @HadesD!)

slouken committed:
Fixed building with the 16.1.4479499 Android toolchain

slouken committed:
Fixed SDL_iconv_string() truncation when handling SDL_ICONV_E2BIG

(cherry picked from commit c369b900199f905c51edb372bd7780f67a544a8c)

slouken committed:
Only convert the result of XLookupString() if it's not already UTF-8

Fixes https://github.com/libsdl-org/SDL/issues/7766

(cherry picked from commit 491ae20d963cff397b5980b31d142d576e74becb)

sezero committed:
fix MSVC build error.

slouken committed:
Don't bother re-encoding Latin1 characters in the ASCII range

(cherry picked from commit 6150b5b3cbde0e592c4ffe822f66aa5f9c90c3d9)

sezero committed:
fixed os/2 mouse relative motion not working (thanks Andrey Vasilkin,)

for issue reported at https://github.com/bitwiseworks/SDL2-os2/issues/5

madebr committed:
git commit -m "cmake: set maximum policy to 3.5

This fixes the following warning when configuring with CMake 3.27+:
```
CMake Deprecation Warning at CMakeLists.txt:3190 (cmake_minimum_required):
Compatibility with CMake < 3.5 will be removed from a future version of
CMake.

Update the VERSION argument <min> value or use a ...<max> suffix to tell
CMake that the project does not need compatibility with older versions.
```

icculus committed:
x11: check if window size/position has changed during SDL_ShowWindow.

Fixes #4216.

(cherry picked from commit 49b5cfa6c5cb9bb8cecce508e12fa7f8e0cc302f)

sezero committed:
further mouse relative motion fix for os/2 (thanks Andrey Vasilkin)

smcv committed:
linux: Reduce magic numbers when mapping gamepad axes

The bitfield `mapped` has two different sets of meanings, depending
whether we're setting up the triggers or the d-pad. Represent them
as symbolic constants rather than opaque integers.

Signed-off-by: Simon McVittie <smcv@collabora.com>
(cherry picked from commit c4d49fadd4500cb522b2a98e5a42d024c566d8bf)

smcv committed:
linux: Improve gamepad mapping heuristic to accept Android conventions

This heuristic for gamepads without a more specific mapping already
tried two incompatible conventions for handling triggers: the Linux
Gamepad Specification uses hat switch 2 for the triggers (for whatever
reason), but the de facto standard set by the drivers for older Xbox
and Playstation controllers represents each trigger as the Z-axis of
the nearest analog stick.

Android documentation encourages Bluetooth gamepad manufacturers to use
a third incompatible convention where the left and right triggers are
represented as the brake and gas pedals of a driving simulator
controller. The Android convention also changes the representation of
the right stick: instead of using X and Y rotation as a second pair
of axes, Android uses Z position as a second horizontal axis, and
Z rotation as a second vertical axis.

Try to cope gracefully with all of these. This will hopefully resolve
the issue described in #5406 (when using unpatched kernels).

Signed-off-by: Simon McVittie <smcv@collabora.com>
(cherry picked from commit cf1dc66e2cfc7a65374c5fea681dd31c50363a2c)

slouken committed:
Fixed reading input from the Razer Atrox Arcade Stick using Windows Gaming Input

(cherry picked from commit 172cfca51332a845abf9c141ae56aa121fe4ada5)

slouken committed:
Only update the serial number if it hasn't already been set

This fixes the serial number for Nintendo Switch Pro, which is queried from the hardware in device initialization, and was later clobbered by the USB string which isn't correct.

(cherry picked from commit 2042e9c4e3cba6ffa9c34abac14828e31365f98b)

slouken committed:
Fixed querying device info on the MOBAPAD M073

The query packet needs to contain valid rumble data in order to be accepted by the controller.

Fixes https://github.com/libsdl-org/SDL/issues/7788

(cherry picked from commit 5490873daa12a57ff4b36090f25f04530323c6e4)

slouken committed:
Make it clear that you can't mix 2D rendering and the window surface API

Also added functions to query and destroy the window surface so you can switch between modes if you want.

See https://github.com/pygame-community/pygame-ce/issues/2190 for more details.

slouken committed:
Fixed build

sezero committed:
SDL_dynapi.c: minor cosmetics.

icculus committed:
cocoa: Warp mouse to center of window before enabling relative mouse.

This prevents the case where the mouse might be at the edge of the
window when enabling relative mode, which confuses macOS, at it
might believe the user is attempting to resize the window.

Fixes #6994.

slouken committed:
Reduce the chance of destroying the joystick mutex while it's in use

Fixes https://github.com/libsdl-org/SDL/issues/7811

(cherry picked from commit 6390165fd4c193631d6780758a4aeec0d02b90eb)

slouken committed:
Fixed n3ds build

(cherry picked from commit d95dbe78bb7a5087ecfb8d301417b1108d868738)

slouken committed:
Fixed enumerating Steam Controllers on iOS

(cherry picked from commit dec0dbff13d4091035209016eb2d0dd82c9aba58)

slouken committed:
Fixed joystick vendor detection in Linux automatic gamepad mapping

(cherry picked from commit 071d1e29dd4f80605932f320aa2e05fda383944c)

slouken committed:
Added automatic mapping for Xbox Elite paddles using the xpadneo driver

We can't actually tell yet whether a controller has paddles, so this code isn't effective, but I'll file an upstream issue and see if we can get that resolved.

(cherry picked from commit b0677f476fa43f4a113b04a959228fd38f95d740)

sezero committed:
fixed build after commit d2372c853843def01b7adb9872cbb89b0d6079ec.

slouken committed:
Added automatic mapping support for Xbox controllers on the 6.x Linux kernels

This automatically adds support for the share button and paddles when present.

(cherry picked from commit db1d4d3d76f5e21b2547463710b513fe0ebd7fad)

slouken committed:
Use the automatic mapping instead of a hardcoded one for Xbox controllers on Linux

This is much more robust and able to dynamically create a mapping for Xbox One S, Xbox Series X, and Xbox Elite 2 controllers.

(cherry picked from commit 9567989eb3ce9c858f0fe76806c5ccad69da89ba)

slouken committed:
Cleanup for previous change, fixing typos, etc.

(cherry picked from commit 883b0f4071687f2e5c8329963f97504e9679394b)

slouken committed:
Removed more Linux Xbox mappings in favor of the automatic mapping

This fixes the Xbox Series X share button on Linux 5.x kernels.

(cherry picked from commit 0f4b15e16b7f07a46db6dc8e651f8c1849d658c5)

slouken committed:
Updated to version 2.27.1 for release candidate

slouken committed:
Added support for the PowerA Fusion Pro Wireless Controller in Bluetooth mode

This controller shows up with a VID/PID of 0, but has full functionality over Bluetooth

(cherry picked from commit cdfc0c5a3314e4e0cd5152feddd8950c7eb797f1)

slouken committed:
Allow switching licensed Nintendo Switch Pro controllers into gyro input mode

(cherry picked from commit 8c95bd814bf1cf0ea1f12aa724938176a7dfd780)

slouken committed:
Fixed third-party Nintendo Switch Pro controllers shutting down when we try to set the home LED.

This fixes the PDP Afterglow Wireless Deluxe Controller.

(cherry picked from commit 0c16f4faf00eb573d051cac7d60ed8dd22e5596a)

slouken committed:
Removed unused function

(cherry picked from commit 73927b09480fd5d664f7077a70a8b6df318cdd2d)

slouken committed:
Added support for the GameSir G4 Pro

We can't read device info or IMU calibration from this controller, and it has no gyro or accelerometer, but is otherwise perfectly functional.

(cherry picked from commit f168f9c81326ad374aade49d1dc46f245b20d07a)

slouken committed:
We don't know in advance whether a device is connected over Bluetooth in SDL2

slouken committed:
Fixed DualSense controllers not being picked up by the HIDAPI driver

The hidraw device may take additional time to get the correct permissions for us to open it. In my tests on Steam Deck hardware, this ranges between 5-8ms.

(cherry picked from commit c6ee9780df4286f66c38f3fa9732daa9afe0a8a3)

slouken committed:
Added support for blitting SDL_PIXELFORMAT_INDEX1LSB

Fixes https://github.com/libsdl-org/SDL/issues/7844

smcv committed:
ibus: Handle error when getting the D-Bus machine ID

It is possible for retrieving the machine ID to fail, either because
dbus was installed incorrectly (machine ID absent or corrupt), or in
32-bit builds, because stat() on the machine ID fails with EOVERFLOW
if it has an out-of-range timestamp or inode number.

dbus has historically treated this as a faulty installation, raising
a warning which by default causes the process to crash. Unfortunately,
dbus_get_local_machine_id() never had a way to report errors, so it has
no alternative for that (bad) error handling.

In dbus >= 1.12.0, we can use dbus_try_get_local_machine_id() to get
the same information, but with the ability to cope gracefully with
errors. ibus won't work in this situation, but that's better than
crashing.

(cherry picked from commit 91198baed40d5709020c3001e9234f4580df696a)

Mitigates: https://github.com/ValveSoftware/steam-for-linux/issues/9605
Signed-off-by: Simon McVittie <smcv@collabora.com>

slouken committed:
Updated to version 2.28.0 for release

slouken committed:
Apply DPI scale to mouse coordinates in SDL_WarpMouseInWindow()

Fixes https://github.com/libsdl-org/SDL/issues/7855

(cherry picked from commit 657c346556f18044dac04957ee7894220e86ac8e)

slouken committed:
Use default sensor calibration if we can't read it from the Nintendo Switch controller

Fixes https://github.com/libsdl-org/SDL/issues/7830

(cherry picked from commit 3694dabe7c16bb8e76f6f4820a1987ce16c4a3c9)
(cherry picked from commit dfbdaca2b8a2bc4c7e1871698066b62b42806a46)

slouken committed:
Removed 100 ms hitch when querying third party Nintendo Switch controllers that don't respond to request for info

(cherry picked from commit 8acf72965031a5101f478eccd69d092001134184)

slouken committed:
SDL_FindFreePlayerIndex() will always return the next available player index

Fixes https://github.com/libsdl-org/SDL/issues/7868

(cherry picked from commit 87e916dd214866af9cfd67eb4618918bc8b0f1b4)
(cherry picked from commit 38619bb28db83bc5844d13d99b3072496adcd498)

german77 committed:
hidapi: switch: Handle MCU input reports

(cherry picked from commit 425062c123f88b6890277905714e02c9375f5b5e)
(cherry picked from commit 950ff0056a74a2cb6f6ff0b5ebaa9c16c305234a)

slouken committed:
Fixed Nintendo Switch Pro controllers that don't have readable user calibration

(cherry picked from commit 9ee1200e6f8fecd394d9355781af40ac90851546)
(cherry picked from commit 170428b107bdc004eb44fa9818eb14c2aa8798d1)

slouken committed:
Don't crash if SDL_MapRGB() and SDL_MapRGBA() are passed a NULL format

(cherry picked from commit fadc4916a9bcc410c2148a91bf0adf742410a4ba)
(cherry picked from commit 0ba93e4aac8232de45e2a6df4d0a42ec53f53ec3)

slouken committed:
Added support for the Nintendo Online Famicom controllers

(cherry picked from commit baa9c57581490bd9cdefabb213e7cb6d0c30b296)
(cherry picked from commit 0f940cb6ce350631b53c77360adffe276cad4660)

slouken committed:
Fixed build

(cherry picked from commit 099e53f3a51d17155063977d5f4b40d0f07f73a2)

slouken committed:
Added Linux mapping for the Logitech Chillstream

Fixes https://github.com/libsdl-org/SDL/issues/7829

(cherry picked from commit 675fc8c38f3f5685d177f0ecf7f43d57a109992e)
(cherry picked from commit fc0854651becb207de28f926e0b84aedbfff8b7b)

icculus committed:
rwops: Use SetFilePointerEx on Windows for appending writes.

Fixes #7900.

(cherry picked from commit 769bf2ebcca8f0cbcde4ec50008942f4eda57f4a)

slouken committed:
Updated to version 2.28.1 for release

giroletm committed:
Added Nintendo Switch patches
Patches by:
- @Cpasjuste
- @WohIstand
- @R-YaTian
- @endrift
- @uyjulian
- @usineur
- @carstene1ns
- @WinterMute
- @giroletm

WinterMute commented:
This PR contains rather a lot of upstream SDL2. The first step to producing a patchset needs to be rebasing the current patches against newer upstream then fixing issues that arise from there.

giroletm commented:
I did include upstream SDL2 commits to have them in the history, rather than a big commit with all changes SDL2 has received since the current version.
Not sure what's the best way to do this. Either way, feel free to build my repo, and see it works, no issues.
Please guide me to how making this PR better, I'm not sure I understand what you mean by "rebasing the current patches against newer upstream then fixing issues".
